authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerHiveAuthorizationPlugin.java

```diff
@@ -0,0 +1,854 @@
+   * Reference: https://github.com/apache/gravitino/issues/4509
+   */
+  @Override
+  public Boolean onRoleDeleted(Role role) throws RuntimeException {
+    return onRoleUpdated(
+        role,
+        role.securableObjects().stream()
+            .map(securableObject -> RoleChange.removeSecurableObject(securableObject))
+            .toArray(RoleChange[]::new));
+  }
+
+  @Override
+  public Boolean onRoleUpdated(Role role, RoleChange... changes) throws RuntimeException {
+    for (RoleChange change : changes) {
+      boolean execResult;
+      if (change instanceof RoleChange.AddSecurableObject) {
+        execResult = doAddSecurableObject((RoleChange.AddSecurableObject) change);
+      } else if (change instanceof RoleChange.RemoveSecurableObject) {
+        execResult =
+            doRemoveSecurableObject(role.name(), (RoleChange.RemoveSecurableObject) change);
+      } else if (change instanceof RoleChange.UpdateSecurableObject) {
+        execResult =
+            doUpdateSecurableObject(role.name(), (RoleChange.UpdateSecurableObject) change);
+      } else {
+        throw new IllegalArgumentException(
+            "Unsupported role change type: "
+                + (change == null ? "null" : change.getClass().getSimpleName()));
+      }
+      if (!execResult) {
+        return Boolean.FALSE;
+      }
+    }
+
+    return Boolean.TRUE;
+  }
+
+  @VisibleForTesting
+  public List<Privilege> getAllPrivileges() {
+    return mapPrivileges.keySet().stream().map(Privileges::allow).collect(Collectors.toList());
+  }
+
+  @Override
+  public Boolean onOwnerSet(MetadataObject metadataObject, Owner preOwner, Owner newOwner)
+      throws RuntimeException {
+    List<Privilege> allPrivileges = getAllPrivileges();
+
+    SecurableObject securableObjects =
+        SecurableObjects.parse(metadataObject.fullName(), metadataObject.type(), allPrivileges);
+
+    AuditInfo auditInfo =
+        AuditInfo.builder()
+            .withCreator(PrincipalUtils.getCurrentUserName())
+            .withCreateTime(Instant.now())
+            .build();
+
+    RoleEntity role =
+        RoleEntity.builder()
+            .withId(0L)
+            .withName(OWNER_ROLE_NAME)
+            .withAuditInfo(auditInfo)
+            .withSecurableObjects(Lists.newArrayList(securableObjects))
+            .build();
+
+    Boolean execResult = onRoleCreated(role);
+    if (!execResult) {
+      return Boolean.FALSE;
+    }
+    if (preOwner != null) {
+      if (preOwner.type() == Owner.Type.USER) {
+        UserEntity userEntity =
+            UserEntity.builder()
+                .withId(0L)
+                .withName(preOwner.name())
+                .withRoleNames(Collections.emptyList())
+                .withRoleIds(Collections.emptyList())
+                .withAuditInfo(auditInfo)
+                .build();
+        execResult = onRevokedRolesFromUser(Lists.newArrayList(role), userEntity);
+        if (!execResult) {
+          return Boolean.FALSE;
```

Comment:
I believe we should log the failure information regarding the steps that failed.